src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -21,8 +24,20 @@ const (
 // handler available from getHandler() and latest ReverseProxy available from
 // getReverseProxy().
 func (client *Client) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	logging.RegisterUserAgent(req.Header.Get("User-Agent"))
+	userAgent := req.Header.Get("User-Agent")
+	logging.RegisterUserAgent(userAgent)
+
+	golog.WithContext(golog.Map{
```

Comment:
Instead of using defer syntax, this uses callback functions.  This has the benefit of making the context boundaries clear and not leaking data.

Also, this version of gls creates nested contexts that contain parent context information but that don't leak their own information back into the parent.